src/pkg/bb/bbinternal/bb.go

```diff
@@ -471,7 +472,16 @@ func (p *Package) rewriteFile(f *ast.File) bool {
 	hasMain := false
 
 	// Change the package name declaration from main to the command's name.
-	f.Name.Name = p.Name
+	// Remove all non-alphanumeric characters except for underscore and ensure
+	// starting with a letter. There are more valid identifiers though.
+	// The Go spec defines the following grammar:
+	//   identifier = letter { letter | unicode_digit } .
+	// See also https://golang.org/ref/spec#Identifiers
+	reg, err := regexp.Compile("[^a-zA-Z0-9_]+")
+	if err != nil {
+		panic(err)
```

Comment:
let's compile this at init time? regexp.MustCompile in an init() or something

Reply:
https://www.reddit.com/r/golang/comments/d6548r/when_and_how_is_it_recommended_to_compile_regular/f0rmb3k/?utm_source=reddit&utm_medium=web2x&context=3 suggests to just declare a global var with regexp.MustCompile, so I'm doing that now, thanks!